### PR TITLE
Java spec update

### PR DESCRIPTION
**Pull request by JonathanGiles**
Splitting Java spec out into separate files and consolidating with the general guidelines. Most of this content has been reviewed previously, however, this is getting close to a final 1.0.0, so it is time to review again (ignoring all the 'draft' comments that will not be resolved immediately). It is recommended to review the generated output here: https://jogiles.z19.web.core.windows.net/java-spec-draft/JavaSpec.html

### CODE REVIEW

**Comment by adrianhall** [comment]
The only thing that stood out is your use in several places of "There exists..." - you should change this to a more active voice

"There exist a number of annotations..." => "A number of annotations exist..."

File: docs/design/java/Namespace.mdk [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+If the client library does not seem to fit into the group list, contact [adparch](https://github.com/Azure/azure-sdk-pr/issues/new/choose) for advice on the most appropriate group.  If you feel your service requires a new group, then open a "Design Guidelines Change" request.
+
+~ MustNot {#java-do-not-leak-impl}
+allow implementation code (i.e. code that does not form part of the public API) to be mistaken as public API. There are two valid arrangements for implementation code, one (or both) of which must be followed:
+
+1. Implementation classes can be placed within a subpackage named `implementation`.
+2. Implementation classes can be made package-private and placed within the same package as the consuming class.
+
+There exists CheckStyle checks that enforce that classes within an `implementation` package are not exposed through public API (that is, all API that is not in an `implementation` package should not return or accept arguments that are types (or subtypes) of a class within an `implementation` package).
```
Comment (adrianhall) [inline docs/design/java/Namespace.mdk]: CheckStyle checks exist.... - also, potentially link to the CheckStyle checks link and describe how the client library writer might run them prior to check in.

File: docs/design/java/Dependencies.mdk [hunk 2]
```diff
@@ -0,0 +1,21 @@
+[INCLUDE=../general/Dependencies.mdk]
+ 
+~ Must {#java-allowed-dependencies}
+only make use of libraries that are listed in the table below. Transitive dependencies of the libraries below, or dependencies that are part of a family of dependencies, such as reactor-netty being a child project of reactor, are transitively permitted.
+
+| Name                                            | Role              | Allowed in public API | Notes |
+|-------------------------------------------------|------------------ |:---------------------:|-------|
+| [Reactor](http://projectreactor.io)             | Reactive library  | Yes                   |       |
+| [Netty](http://netty.io)                        | HTTP client       | Yes                   |       |
```
Comment (joshfree) [inline docs/design/java/Dependencies.mdk]: have we standardized on a minimum version of netty?  is it 4.1?

File: docs/design/java/Versioning.mdk [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+have a qualifier of `-preview.N` for all preview releases.
+~
+
+~ Must {#java-preview-version-next-GA}
+have the version number represent the version intended for the next GA release. For example, a preview release of `1.0.2-beta2` should be read to mean that it is the second beta release of version 1.0.2, and that this release will be superceded by the GA release of 1.0.2 when it becomes available.
+~
+
+### GA Releases
+
+Anything that is not a snapshot or preview release should be considered an official GA release. Official releases must be done only once per version increment, and must only be done through the engineering system pipeline.
```
Comment (joshfree) [inline docs/design/java/Versioning.mdk]: nit: should there be a link to an official language-agnostic document on GA releases included here?

File: docs/design/java/Namespace.mdk (hunk 1 above)
Comment (JonathanGiles) [reply docs/design/java/Namespace.mdk]: Active voice used. CheckStyle instructions are included in spec a few sections above (at the start of the Java tooling section).

File: docs/design/java/Dependencies.mdk (hunk 2 above)
Comment (JonathanGiles) [reply docs/design/java/Dependencies.mdk]: Yes, we have specified 4.1.33.Final as the current version of Netty that we use. It is purposefully not specified in this document as there is already a source of truth - the parent pom.xml file that all modules refer to as their parent pom. You can see on line 118 the [netty dependency](https://github.com/Azure/azure-sdk-for-java/blob/master/parent/pom.xml#L118). I will provide a link from the spec to this file.

**Pull request merged**